### PR TITLE
Replace custom Awk and remove related tests

### LINKED ISSUE #1700: Replace pkg.util.sys.Awk() with strings.Fields()
**Feature Request**

**Are there any similar features already existing:**
No

**What should the feature do:**
Replace the custom Awk() function and all of its tests with the [strings.Fields()](https://golang.org/pkg/strings/#Fields) function found in the standard library. We don't rely on the 0th index special case and there's not a clear reason for keeping this function around.

**What would be solved through this feature:**
A leaner and cleaner code base.

**Does this have an impact on existing features:**
No, we don't rely on any special cases.

### PR DESCRIPTION
**Pull request by checkaayush**
Signed-off-by: Aayush Sarva <checkaayush@gmail.com>

Description of your changes: Replaced custom Awk function with `strings.Fields` and removed related tests along with some obsolete functions. 

Which issue is resolved by this Pull Request:
Resolves #1700 

Checklist:
- [x] Documentation has been updated, if necessary.
- [x] Pending release notes updated with breaking and/or notable changes, if necessary.
- [x] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.

### CODE REVIEW

File: pkg/util/sys/device.go [hunk 1]
```diff
@@ -263,7 +267,11 @@ func GetDeviceFromMountPoint(mountPoint string, executor exec.Executor) (string,
 	}
 
 	searchFor := fmt.Sprintf("on %s ", mountPoint)
-	device := Awk(Grep(output, searchFor), 1, " ")
```
Comment (travisn) [inline pkg/util/sys/device.go]: looks like `GetDeviceFromMountPoint` is also obsolete, just has a unit test referencing it

File: tests/framework/utils/helm_helper.go [hunk 2]
```diff
@@ -62,7 +62,12 @@ func (h *HelmHelper) GetLocalRookHelmChartVersion(chartName string) (string, err
 	}
 	cd := strings.Replace(sys.Grep(result, chartName), "\t", " ", 2)
 
-	return sys.Awk(cd, 2, " "), nil
+	version := ""
+	slice := strings.Fields(cd)
+	if len(slice) >= 2 {
+		version = slice[1]
+	}
+	return version, nil
```
Comment (travisn) [inline tests/framework/utils/helm_helper.go]: should we actually return an error if the field is not found?

File: pkg/util/sys/device.go [hunk 3]
```diff
@@ -250,7 +250,11 @@ func GetDeviceMountPoint(deviceName string, executor exec.Executor) (string, err
 	}
 
 	searchFor := fmt.Sprintf("^/dev/%s on", deviceName)
-	mountPoint := Awk(Grep(output, searchFor), 3, " ")
```
[the remark this answers is not in the document]
Comment (checkaayush) [reply pkg/util/sys/device.go]: Yes, missed that. Removed.

File: pkg/util/sys/device.go (hunk 1 above)
Comment (checkaayush) [reply pkg/util/sys/device.go]: Removed.

File: tests/framework/utils/helm_helper.go (hunk 2 above)
Comment (checkaayush) [reply tests/framework/utils/helm_helper.go]: Done. Also, removed the replacement of tab character since `strings.Fields` handles tab characters as well.

**Comment by travisn** [comment]
@checkaayush looks good, thanks for the updates. Just a couple minor suggestions:
- Could you squash the two commits? 
- No need to incude the fix issue in the PR title

**Comment by checkaayush** [comment]
@travisn Done. Won't include fix issue in PR's title from now on. 

Just curious from project maintenance perspective, wouldn't it be better for the maintainer to always squash and merge any PR (giving the ability to choose the commit msg that goes into master)? I might be missing something here.

**Comment by travisn** [comment]
@checkaayush yes that would be ideal to squash and merge for you, good question. I’ll look at enabling that option, it is currently disabled.

**Pull request merged**